libs/blocks/global-navigation/utilities/keyboard/popup.js

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable class-methods-use-this */
+import {
+  getNextVisibleItem,
+  getPreviousVisibleItem,
+  getOpenPopup,
+  selectors,
+} from './utils.js';
+import { closeAllDropdowns } from '../utilities.js';
+
+const getState = ({ e } = {}) => {
+  const popupEl = getOpenPopup();
+  if (!popupEl) return {};
+  const popupItems = [...popupEl.querySelectorAll(selectors.popupItems)];
+  const curr = popupItems.findIndex((el) => el === e.target);
+  const prev = getPreviousVisibleItem(curr, popupItems);
+  const next = getNextVisibleItem(curr, popupItems);
+  const column = document.activeElement.closest(selectors.column);
+  const visibleColumns = [...popupEl.querySelectorAll(selectors.column)];
+  const currentColumn = visibleColumns.findIndex((node) => node.isEqualNode(column));
+  const prevColumn = visibleColumns[currentColumn - 1] || -1;
+  const nextColumn = visibleColumns[currentColumn + 1] || -1;
+
+  return {
+    popupItems,
+    curr,
+    prev,
+    next,
+    prevColumn,
+    nextColumn,
+  };
+};
+class Popup {
+  constructor({ mainNav }) {
+    this.mainNav = mainNav;
+    this.addEventListeners();
+    this.desktop = window.matchMedia('(min-width: 900px)');
+  }
+
+  open({ focus } = {}) {
+    const popupItems = [...(getOpenPopup()?.querySelectorAll(selectors.popupItems) || [])];
+    if (!popupItems.length) return;
+    const first = getNextVisibleItem(-1, popupItems);
+    const last = getPreviousVisibleItem(popupItems.length, popupItems);
+    if (focus === 'first') popupItems[first].focus();
+    if (focus === 'last') popupItems[last].focus();
+  }
+
+  addEventListeners = () => {
+    document.querySelector('header').addEventListener('keydown', (e) => {
+      const popupEl = getOpenPopup();
+      if (!e.target.closest(selectors.popup) || !popupEl || !this.desktop.matches) return;
+      e.preventDefault();
+      e.stopPropagation();
+      const { popupItems, prev, next, prevColumn, nextColumn } = getState({ e });
+
+      if (e.shiftKey && e.code === 'Tab') {
+        if (prev === -1) {
+          this.mainNav.items[this.mainNav.curr].focus();
+          return;
+        }
+        popupItems[prev].focus();
+        return;
+      }
```

Comment:
Why not include this in the `switch` statement for `Tab`?

Reply:
I've been on that code way too long, thanks for the second pair of eyes! It didn't even come into my mind at this point anymore. Moved them all accordingly